### ISSUE TITLE
add support for Commons images to Vanderbot

### ISSUE THREAD

**Issue by baskaufs** [HeardLibrary/linked-data#20]
Testing reveals that the snak JSON for a commons image looks like this:

```
"P18": [
  {
    "mainsnak": {
      "snaktype": "value",
      "property": "P18",
      "hash": "c99db2661b192fd22f5dff1a532b1d6eb9433ef9",
      "datavalue": {
        "value": "Ace The Wonder Dog.jpg",
        "type": "string"
      },
      "datatype": "commonsMedia"
    },
    "type": "statement",
    "id": "Q15397819$0d43e90b-4963-b88f-49c4-fc42afe5b606",
    "rank": "normal"
  }
]
```

When queried via SPARQL, the returned value is:

```
http://commons.wikimedia.org/wiki/Special:FilePath/Ace%20The%20Wonder%20Dog.jpg
```

**Comment by baskaufs** [comment]
Here's the fix, around line 1270:

```
                        print('Property column: ', propColumnHeader, ', Property ID: ', propertyId, ' Value type: string')

                        if propertyId == 'P18': # special handling for images
                            datatype = 'commonsMedia'
                        else:
                            dataype = 'string'

                        propertiesEntityOrLiteral.append('literal')
                        propertiesTypeList.append(datatype)
                        propertiesValueTypeList.append('string')
                        propertiesLangList.append('')
```

**Comment by baskaufs** [comment]
To get the raw filename from a SPARQL query, use:

```
from urllib.parse import unquote
pieces = string.split('/')
image_name_encoded = pieces[len(pieces)-1]
print(unquote(image_name_encoded))
```

**Comment by baskaufs** [comment]
fixed in Version 1.8
